source/main.py

```diff
@@ -0,0 +1,307 @@
+    parser.add_argument("--data_path", type=str, default="../data/preprocessed/fullcorpus.feather",
+                        help="Input data path")
+    parser.add_argument("--test_data_path", type=str, default="",
+                        help="Input test data path")
+    parser.add_argument('--batch_size', type=int, default=16,
+                        help='input batch size for training (default: 64)')
+    #parser.add_argument('--val_batch_size', type=int, default=1000,
+    #                    help='input batch size for validation (default: 1000)')
+    parser.add_argument('--epochs', type=int, default=10,
+                       help='number of epochs to train (default: 10)')
+    parser.add_argument('--lr', type=float, default=2e-5,
+                        help='learning rate (default: 0.01)')
+    parser.add_argument('--weight_decay', type=float, default=0.01,
+                        help='SGD momentum (default: 0.5)')
+    #parser.add_argument('--momentum', type=float, default=0.5,
+    #                    help='SGD momentum (default: 0.5)')
+    parser.add_argument('--log_interval', type=int, default=20,
+                        help='how many batches to wait before logging training status')
+    parser.add_argument("--log_dir", type=str, default="../data/logs",
+                        help="log directory for Tensorboard log output")
+    parser.add_argument("--checkpoint_dir", type=str, default="../data/model_checkpoints/",
+                        help="log directory for Tensorboard log output")
+    parser.add_argument("--n_files", type=int, default=2,
+                        help="semcor number of files"),
+    parser.add_argument("--model", type=str, default='bert') # can be xlnet as well
+    parser.add_argument("--model_type", type=str, default='bert-base-uncased',
+                        help="bert model: default is bert-base-uncased")
+    parser.add_argument("--token_layer", type=str, default='token-cls',
+                        help="bert token layer type: default is token-cls")
+    parser.add_argument("--weak_supervision", type=bool, default=False,
+                        help="Enable context gloss weak supervision")
+    parser.add_argument("--optimize_gpu_mem", type=bool, default=False,
+                        help="Enable non_blocking argument in pytorch to speedup GPU memory transfers")
+    parser.add_argument("--num_workers", type=int, default=0,
+                        help="Enable non_blocking argument in pytorch to speedup GPU memory transfers")
+    parser.add_argument("--preprocess_inputs", type=bool, default=False,
+                        help="Preprocess input data (Tokenize and generate input embeddings)")
+    args = parser.parse_args()
+                        
+             
+    print('Running with {}'.format(args))
+    print()
+    # ## Process Data
+    print('Preprocessing data')      
+    dl = gen_dataloader(args.data_path,args.test_data_path,args.batch_size,
```

Comment:
When I run main.py, I get the following error: 

Using TensorFlow backend.
Running with Namespace(batch_size=16, checkpoint_dir='../data/model_checkpoints/', data_path='../data/preprocessed/fullcorpus.feather', epochs=10, log_dir='../data/logs', log_interval=20, lr=2e-05, model='bert', model_type='bert-base-uncased', n_files=2, num_workers=0, optimize_gpu_mem=False, preprocess_inputs=False, test_data_path='', token_layer='token-cls', weak_supervision=False, weight_decay=0.01)

Preprocessing data
Traceback (most recent call last):
  File "main.py", line 288, in <module>
    tokenizer_type = args.model_type,
  File "/mnt/fs03/home/nlopatina/NLP_word_sense_disambiguation/source/dataloaders/dataloader_utils.py", line 57, in gen_dataloader
    train_dataset = read_data_to_dataframe(_train_path)
  File "/mnt/fs03/home/nlopatina/NLP_word_sense_disambiguation/source/dataloaders/dataloader_utils.py", line 24, in read_data_to_dataframe
    return pd.read_feather(_path,**kwargs)
  File "/home/nlopatina/.local/lib/python3.6/site-packages/pandas/util/_decorators.py", line 208, inwrapper
    return func(*args, **kwargs)
  File "/home/nlopatina/.local/lib/python3.6/site-packages/pandas/io/feather_format.py", line 119, in read_feather
    return feather.read_feather(path, columns=columns, use_threads=bool(use_threads))
  File "/home/nlopatina/.local/lib/python3.6/site-packages/pyarrow/feather.py", line 213, in read_feather
    reader = FeatherReader(source)
  File "/home/nlopatina/.local/lib/python3.6/site-packages/pyarrow/feather.py", line 40, in __init__
    self.open(source)
  File "pyarrow/feather.pxi", line 80, in pyarrow.lib.FeatherReader.open
  File "pyarrow/io.pxi", line 1323, in pyarrow.lib.get_reader
  File "pyarrow/io.pxi", line 1312, in pyarrow.lib._get_native_file
  File "pyarrow/io.pxi", line 780, in pyarrow.lib.memory_map
  File "pyarrow/io.pxi", line 743, in pyarrow.lib.MemoryMappedFile._open
  File "pyarrow/error.pxi", line 87, in pyarrow.lib.check_status
pyarrow.lib.ArrowIOError: Failed to open local file '../data/preprocessed/fullcorpus.feather', error: No such file or directory